### PR TITLE
Refactors execute saving and retrievial of table data

### PR DESCRIPTION
**Pull request by diegostruk1**
- Uses FileStorage API to save json intermediate node data.
- Adds a field workspace name to Workflow object. 
- Static method for deriving the name of the file to be written/read.
* Probably need to discuss how the workflow name will be populated in front-end and passed to back-end.

### CODE REVIEW

File: pyworkflow/pyworkflow/workflow.py [hunk 1]
```diff
@@ -13,9 +13,11 @@ class Workflow:
         file_path: Location of a workflow file
     """
 
-    def __init__(self, graph=nx.DiGraph(), file_path=None):
+    def __init__(self, graph=nx.DiGraph(), file_path=None, workflow_name='a-name'):
+        #TODO: need to discuss a way to generating the workflow name. For now passing a default name.
```
Comment (reddigari) [inline pyworkflow/pyworkflow/workflow.py:17]: We can use the diagram UUID when hitting the `workflow/new` endpoint. Either in a querystring param, or by making the endpoint POST

File: pyworkflow/pyworkflow/workflow.py [hunk 2]
```diff
@@ -69,6 +71,9 @@ def update_or_add_node(self, node: Node):
 
         return
 
+    def workflow_name(self):
```
Comment (reddigari) [inline pyworkflow/pyworkflow/workflow.py:74]: Pythonic way is `get_workflow_name`

File: vp/node/views.py [hunk 3]
```diff
@@ -228,11 +230,8 @@ def execute_node(request, node_id):
                      })
 @api_view(['GET'])
 def retrieve_data(request, node_id):
-    # TODO: currently hard coding the name of the file as workflow + node id;
-    #       will need to change the word workflow for the workflow name. Will also need to add some exceptions.
-    #
-    file_name = 'workflow-' + str(node_id)
-
+    # TODO: need to add validation probably
+    file_name = Workflow.generate_file_name(request.pyworkflow.workflow_name(), node_id)
```
Comment (reddigari) [inline vp/node/views.py:234]: I'm not familiar with the FileStorage stuff, but do we not need `fs.open(...)` to read the file from /tmp?

File: pyworkflow/pyworkflow/workflow.py (hunk 1 above)
Comment (diegostruk1) [reply pyworkflow/pyworkflow/workflow.py:17]: @reddigari that sounds like a good idea. I'll go ahead with that.

File: pyworkflow/pyworkflow/workflow.py (hunk 2 above)
Comment (diegostruk1) [reply pyworkflow/pyworkflow/workflow.py:74]: Fixed

File: vp/node/views.py (hunk 3 above)
Comment (diegostruk1) [reply vp/node/views.py:234]: It worked fine as is. But I'll give it a try with fs, since seems to be a better way of doing it anyway.

**Pull request merged**